crates/sncast/src/starknet_commands/verify/base.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use camino::Utf8PathBuf;
+use reqwest::StatusCode;
+use serde::Serialize;
+use sncast::response::structs::VerifyResponse;
+use starknet::core::types::Felt;
+use std::ffi::OsStr;
+use walkdir::WalkDir;
+
+fn read_workspace_files(
+    workspace_dir: &Utf8PathBuf,
+) -> Result<serde_json::Map<String, serde_json::Value>> {
+    // Read all files name along with their contents in a JSON format
+    // in the workspace dir recursively
+    // key is the file name and value is the file content
+    let mut file_data = serde_json::Map::new();
+
+    // Recursively read files and their contents in workspace directory
+    for entry in WalkDir::new(workspace_dir.clone()).follow_links(true) {
+        let entry = entry?;
+        let path = entry.path();
+        if path.is_file() {
+            if let Some(extension) = path.extension() {
+                if extension == OsStr::new("cairo") || extension == OsStr::new("toml") {
```

Comment:
Can we include the following files?
- *.md (e.g, `README.md`)
- `LICENSE`
- `.tool-versions`